src/mid.cpp

```diff
@@ -309,7 +319,20 @@ bool CmidPlayer::load(const std::string &filename, const CFileProvider &fp)
             if (s[1]=='D' && s[2]=='L') good=FILE_LUCAS;
             break;
         case 'M':
-            if (s[1]=='T' && s[2]=='h' && s[3]=='d') good=FILE_MIDI;
+            if (s[1]=='T' && s[2]=='h' && s[3]=='d')
+            {
+                /* "MThd"              header-chunk
+                 * 0x00 0x00 0x00 0x06 header size
+                 * 0x00 0x0n           midi file type: 0=single-track format, 1=multiple-track format, 2=multiple-song format
+                 * 0xnn 0xnn           track-number
+                 * 0xnn 0xnn           tempo
+                 */
+                unsigned char s2[4];
+                f->readString((char *)s2, 4);
+                midi_type=u16_unaligned(s2 + 2);
+                good=FILE_MIDI;
+                midiprintf ("General MIDI type: %d\n", midi_type);
```

Comment:
Reading MIDI type-1 this way results in variable getting wrong value (256 instead of 1), because the value is stored in Big Endian format. Better rewrite this part with binio functions.